src/pulselistener/pulselistener/listener.py

```diff
@@ -141,25 +148,29 @@ def list_differential(self):
             logger.error('Invalid webhook parameters', path=request.path_qs)
             raise web.HTTPBadRequest(text='Invalid webhook parameters')
 
+        # Check if the revision if public, and retry a few times
+        # to give some time to the BMO daemon to update the revision
+        public = False
+        for step in range(self.phabricator_retries):
+            if self.is_public_revision(int(revision_id)):
+                public = True
+                break
+            else:
+                sleep_time = self.phabricator_sleep * (1 + (step / self.phabricator_retries))
+                logger.info('Sleeping until next retry', seconds=sleep_time)
+                await asyncio.sleep(sleep_time)
```

Comment:
Waiting here could cause a problem if we end up waiting more than 30 seconds (Heroku will reply to Phabricator with a timeout error), which might very well happen. Not sure how to solve this, given that we want to reply HTTPForbidden for secure revisions (if we didn't want to do it, we could have put the requests in a queue handled in another thread and just always replied 200 to Phabricator).